File: ebook_front/src/services/AdminService.js
import {ip} from "../App";


export const getUsers=(callback)=>{

    fetch(ip+'/getUsers') // 发送fetch请求获取联系人信息的接口地址
        .then(response => response.json())
        .then(data => {
           callback(data);
        })
        .catch(error => {
            console.error('Error fetching contacts:', error);
        });

}

export const banUser=(params,callback)=>{

    fetch(ip+'banUser?'+params.toString()) // 发送fetch请求获取联系人信息的接口地址
        .then(response => response.json())
        .then(data => {
           callback(data);
        })
        .catch(error => {
            console.error('Error fetching contacts:', error);
        });

}

export const unbanUser=(params,callback)=>{

    fetch(ip+'/unbanUser?'+params.toString()) // 发送fetch请求获取联系人信息的接口地址
        .then(response => response.json())
        .then(data => {
            callback(data);
        })
        .catch(error => {
            console.error('Error fetching contacts:', error);
        });

}

export const AdminFilterOrderBook=(params,callback)=>{

    fetch(ip+'/AdminFilterOrderBook?'+params.toString())
        .then(response => response.json())
        .then((data) => {
           callback(data);
        })
}

export const AdminFilterOrderDate=(params,callback)=>{

    fetch(ip+'/AdminFilterOrderDate?'+params.toString())
        .then(response => response.json())
        .then((data) => {
           callback(data);
        })
}

export const getSale=(params,callback)=>{

    fetch(ip+'/getSale?'+params.toString())
        .then(response => response.json())
        .then((data) => {
           callback(data);
        })
}

export const getConsumption=(params,callback)=>{

    fetch(ip+'/getConsumption?'+params.toString())
        .then(response => response.json())
        .then((data) => {
            callback(data);
        })

}

export const getBook=(params,callback)=>{

    fetch(ip+'/getBook?'+params.toString())
        .then(response => response.json())
        .then((data) => {
            callback(data);
        });

}

export const deleteBook=(params,callback)=>{

    fetch(ip+'/deleteBook?'+params.toString())
        .then(response => response.json())
        .then((data) => {
            callback(data);
        })
}
export const editBook=(params,callback)=>{

    fetch(ip+'/editBook?'+params.toString())
        .then(response => response.json())
        .then((data) => {
            callback(data);

        })
}

export const addBook=(params,callback)=>{

    fetch(ip+'/addBook?'+params.toString())
        .then(response => response.json())
        .then((data) => {
            callback(data);
        })


}

File: ebook_back/mainservice/src/main/java/com/example/mainservice/Service/AdminService.java
package com.example.mainservice.Service;

import com.example.mainservice.Entity.Book;
import com.example.mainservice.Entity.JSON.adminConsume;
import com.example.mainservice.Entity.JSON.adminSale;
import com.example.mainservice.Entity.Orders;
import com.example.mainservice.Entity.User;
import org.springframework.web.bind.annotation.RequestParam;

import java.text.ParseException;
import java.util.List;

public interface AdminService {
    List<User> getUsers();
    void deleteBook(@RequestParam int bid);
    void editBook(int bid,String name,
                  String author, String image,
                  String isbn, String inventory,
                  String type, String price,
                  String description
    );

    List<Book> addBook(String name, String author,
                 String image, String isbn,
                 String inventory, String type,
                 String price, String description
    );
    /////////////////
    void banUser( int uid);

    void unbanUser( int uid);
    List<Orders> AdminFilterOrderDate( String start,  String end) throws ParseException;
    List<Orders> AdminFilterOrderBook( String bookName);

    List<adminSale> getHotSell( String start,  String end)throws ParseException;
    List<adminConsume> getConsumption( String start,  String end)throws ParseException;
}
